File: src/shared/domain/console.ts
import type { NodePath } from './tree';

/**
 * One `execute()` batch (§8.14, P5.5 D-plan): `path` binds the batch to a connection and,
 * optionally, a default database/schema; `statements` is the pre-split list from
 * `sql-split.ts` — one call covers both "Run statement" (a one-element array) and "Run all".
 */
export interface ConsoleRequest {
  path: NodePath;
  statements: string[];
}

File: shell/internal/storage/model/console.go
package model

// ConsoleRequest is domain/console.ts's ConsoleRequest — one execute() batch: path binds it to a
// connection and, optionally, a default database/schema; Statements is the pre-split list from
// sql-split.ts (one call covers both "Run statement" and "Run all").
type ConsoleRequest struct {
	Path       NodePath
	Statements []string
}
